connector_magento/components/backend_adapter.py

```diff
@@ -153,9 +153,20 @@ def call(self, method, arguments, http_method=None, storeview=None):
             # record(method, arguments, result)
             return result
         except (socket.gaierror, socket.error, socket.timeout) as err:
-            raise NetworkRetryableError(
-                'A network error caused the failure of the job: '
-                '%s' % err)
+            if err.response is not None:
+                if err.response.headers.get('Content-Type') == 'application/json':
+                    error_text = err.response.json()
+                else:
+                    error_text = err.response.text
```

Comment:
I don't think we expect Magento to return regular text in this case, so you can skip this possibility.